src/ifElse.ts

```diff
@@ -0,0 +1,35 @@
+import { fold } from 'fp-ts/lib/boolean';
+import { pipe } from 'fp-ts/lib/pipeable';
+
+function _ifElse<A, R>(predicate: (a: A) => boolean, whenTrue: (a: A) => R, whenFalse: (a: A) => R, a: A): R {
+  return pipe(
+    a,
+    predicate,
+    fold(
+      () => whenTrue(a),
+      () => whenFalse(a)
+    )
+  );
+}
+
+/**
+ * Similar to [R.ifElse](https://ramdajs.com/docs/#ifElse). Creates a function that will
+ * process either the onTrue or the onFalse function depending upon the result of the condition predicate..
+ *
+ * @since 0.1.7
+ */
+
+export function ifElse<A, R>(predicate: (a: A) => boolean, whenTrue: (a: A) => R, whenFalse: (a: A) => R): (a: A) => R;
```

Comment:
we should also consider adding the additional overload as suggested in #32 I think:
```
ifElse<A, B, R>(predicate: (a: A) => a is B, whenTrue: (a: B) => R, whenFalse: (a: A) => R, a: A): R
```

Also, if I'm not wrong, all the other functions are manually curried to provide all of the possible combinations of parameters application, e.g. both `ifElse(predicate)(whenTrue, whenFalse)(a)` and `ifElse(predicate, whenTrue)(whenFalse, a)`
It can become quite cumbersome to define the signature in this case (4 parameters), we should consider shortening it like done e.g. in `clamp`: https://github.com/giogonzo/fp-ts-ramda/blob/master/src/clamp.ts#L12-L19

Reply:
I tried the second signature as well but ts complained about R not being B and things like that 😅